### ISSUE TITLE
Dialyzer warnings

### ISSUE THREAD

**Issue by bigardone** [sascha-wolf/ex_union#1]
👋🏼 Hi there. Thanks for sharing this library 🙌🏼 .
I'm playing around with it to implement some basic union types. However, Elixir's LD Dialyzer is complaining about overloaded contracts and invalid specifications: 

```elixir
defmodule MyApp.Result do
  import ExUnion

  defunion(ok(value) | error(value))
end
```

```
Overloaded contract for 'Elixir.MyApp.Result.Error':new/1 has overlapping domains; such contracts are currently unsupported and are simply ignored (ElixirLS Dialyzer)
──────────────────────────────────────────────────────────────────────────────
Overloaded contract for 'Elixir.MyApp.Result.Ok':new/1 has overlapping domains; such contracts are currently unsupported and are simply ignored (ElixirLS Dialyzer)
──────────────────────────────────────────────────────────────────────────────
Invalid type specification for function 'Elixir.MyApp.Result':'__union__'/0. The success typing is 
          () ->
             #{'__struct__' := 'Elixir.ExUnion.Definition',
               'module' := 'Elixir.MyApp.Result',
               'name' := <<_:48>>,
               'types' :=
                   [#{'__struct__' := 'Elixir.ExUnion.Definition.Type',
                      'fields' := [any(), ...],
                      'module' :=
                          'Elixir.MyApp.Result.Error' |
                          'Elixir.MyApp.Result.Ok',
                      'name' := 'error' | 'ok'},
                    ...]} (ElixirLS Dialyzer)
```

Are you also experiencing this?

**Comment by sascha-wolf** [comment]
Thank you for the report. I knew I forgot something  - that is checking the generated type-specs. 😄

I'll check it out over the week. 👍

**Comment by bigardone** [comment]
Awesome! 🙌🏼

**Comment by sascha-wolf** [comment]
Resolved in v0.1.1 ([Link to Hex](https://hex.pm/packages/ex_union/0.1.1))

**Comment by bigardone** [comment]
Awesome @sascha-wolf, thank you 🙌🏼 !
